actions/publish-github-pages/action.yml

```diff
@@ -1,10 +1,106 @@
 name: 'Publish GitHub Pages'
-description: 'Publish the Hugo-based CloudPosse docs website to GitHub Pages.'
+description: 'Publish the Hugo-based Cloud Posse docs website to GitHub Pages.'
+inputs:
+  content:
+    description: 'Comma-separated list of directories in the top level of the repo that contains documentation'
+    required: false
+  css:
+    description: 'Custom CSS to customize the look and feel'
+    required: false
+  debug:
+    description: 'Toggle to turn on debug functionality'
+    required: false
+  hugo_url:
+    description: 'URL of the Hugo site after deployment'
+    required: false
+  hugo_publish_dir:
+    description: 'Directory in the repo that GitHub Pages will deploy to'
+    required: false
+  hugo_config:
+    description: 'Location of to-be-written Hugo config file (actual location not important)'
+    required: false
+  htmltest_config:
+    description: 'Location to write the htmltest config file (actual location not important)'
+    required: false
+  github_token:
+    description: 'GitHub token'
+    required: false
+  github_actor:
+    description: 'GitHub Actor'
+    required: false
+  github_pages_hugo_path:
+    description: 'Location where the repo containing the Hugo files will be cloned'
+    required: false
+  github_pages_pull_branch:
+    description: 'The branch of the repo which contains the documentation'
+    required: false
+  staging_dir:
+    description: 'Directory to collate all docs and Hugo configuration files'
+    required: false
 runs:
-#  using: 'docker'
-#  image: 'Dockerfile'
   using: 'composite'
   steps:
-    - run: ${{ github.action_path }}/entrypoint.sh
+    - id: checkout-repos
+      run: |
+           pip install GitPython pyyaml
+           python -c 'from checkout_and_collate import *; checkout_repos()'
       shell: bash
-#    - run: ./actions/publish-github-pages/entrypoint.sh
+      env:
+        PYTHONPATH: ${{ github.action_path }}
+        GITHUB_PAGES_PUSH_PATH: ${{ github.action_path }}/${{ env.GITHUB_PAGES_DIRECTORY }}
+        INPUT_GITHUB_PAGES_PULL_BRANCH: ${{ inputs.GITHUB_PAGES_PULL_BRANCH }}
+        INPUT_GITHUB_PAGES_HUGO_PATH: ${{ inputs.GITHUB_PAGES_HUGO_PATH }}
+        INPUT_DEBUG: ${{ inputs.DEBUG }}
+        INPUT_STAGING_DIR: ${{ inputs.STAGING_DIR }}
+    - id: append-custom-css
+      run: |
+           export CSS=${CSS:-$INPUT_CSS}
+           export GITHUB_PAGES_HUGO_PATH=${GITHUB_PAGES_HUGO_PATH:-$INPUT_GITHUB_PAGES_HUGO_PATH}
+           echo -e "$CSS" >> ${GITHUB_PAGES_HUGO_PATH}/themes/cloudposse/src/scss/custom.scss
```

Comment:
```
id: append-custom-css
  run: |
    yq ... ${{input.config}} ... config.yaml
    
```

Reply:
We should deep merge the YAML configs, but replace values (e.g. lists) do not append to lists.